pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -41,12 +41,12 @@ _no_input = object()
 
 cdef inline object create_timestamp_from_ts(int64_t value,
                                             npy_datetimestruct dts,
-                                            object tz, object freq):
+                                            object tz, object freq, bint fold):
```

Comment:
This again seems like an unnecessary degree of freedom, since `fold` can be easily calculated from the other values provided here, and you'll probably want to do that *anyway* because by adding this unnecessary degree of freedom, you introduce a potential error condition: `value` + `tz` may resolve to something with a different value of `fold` than is passed to this function. 

Admittedly the same argument can be made about the `value` and `dts` components, though - `value` + `tz` should be entirely sufficient to construct `dts` and `fold`.

I can see it if this function is only ever called by things which are guaranteed to generate harmonious values for `dts` and `fold`, but if that's the case it would be a good idea to document that this function is over-specified and that no validation is done.

Reply:
I think we can resolve this conversation, since this actually does seem to be a thin wrapper around the `Timestamp` constructor?

Reply:
Yes, you are right.